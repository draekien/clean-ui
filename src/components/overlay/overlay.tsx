// eslint-disable-next-line tsdoc/syntax
/** @jsxImportSource theme-ui */
import React from 'react';
import { overlayCss } from './overlay.styles';

export const Overlay: React.FC<React.HTMLAttributes<HTMLDivElement>> = ({
  children,
  ...rest
}) => (
  <div sx={overlayCss} {...rest}>
    {children}
  </div>
);
