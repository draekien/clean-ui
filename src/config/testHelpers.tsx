import React, { ReactNode } from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '../components/theme/theme.provider';

export const themedRender = (element: ReactNode) =>
  render(<ThemeProvider>{element}</ThemeProvider>);
