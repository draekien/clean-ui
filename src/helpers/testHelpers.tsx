import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '../components/theme/theme.provider';

export const themedRender = (element: React.ReactNode) => {
  return render(<ThemeProvider>{element}</ThemeProvider>);
};

export const hexToRgb = (hex: string) => {
  hex = hex.replace(/[^0-9A-F]/gi, '');
  const bigint = parseInt(hex, 16);
  const r = (bigint >> 16) & 255;
  const g = (bigint >> 8) & 255;
  const b = bigint & 255;

  return `rgb(${r}, ${g}, ${b})`;
};
