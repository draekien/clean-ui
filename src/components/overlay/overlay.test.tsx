import React from 'react';
import { themedRender } from '../../helpers/testHelpers';
import { Overlay } from './overlay';

describe('Overlay component', () => {
  it('should match snapshot', () => {
    const { container } = themedRender(<Overlay>test</Overlay>);

    expect(container).toMatchSnapshot();
  });
});
