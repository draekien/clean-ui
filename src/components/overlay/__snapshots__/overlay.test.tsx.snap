// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay component should match snapshot 1`] = `
<div>
  <div
    class="css-1hdj22u-Overlay"
  >
    test
  </div>
</div>
`;
